src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -284,40 +310,87 @@ export default class PhotoField extends React.Component {
   }
 
   onZoom = (e) => {
+    // Cropper returns the attempted zoom value
     const zoomValue = e.detail.ratio;
-    if (zoomValue < MAX_RATIO && zoomValue > MIN_RATIO) {
-      this.setState({ zoomValue });
-      this.updateBoundaryWarningAndButtonStates();
 
-      const moveBoundariesMet = getMoveBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
-      const zoomWarn = zoomValue >= WARN_RATIO;
+    // check if zoom is out of bounds
+    let zoomToBoundaryRatio;
+    if (zoomValue < this.state.minRatio) {
+      zoomToBoundaryRatio = this.state.minRatio;
+    } else if (zoomValue > this.state.maxRatio) {
+      zoomToBoundaryRatio = this.state.maxRatio;
+    }
 
-      this.setState({
-        warningMessage: makeCropBoundaryWarningMessage({ zoomWarn, ...moveBoundariesMet }),
-        ...makeMoveButtonsEnabledStates(moveBoundariesMet)
-      });
-    } else {
-      e.preventDefault();
+    // force zoom within zoom bounds
+    if (zoomToBoundaryRatio) {
+      this.refs.cropper.zoomTo(zoomToBoundaryRatio); // force zoom within constraints
+      e.preventDefault(); // prevents bad zoom attempt
+      return; // don't update state until the subsequent zoom attempt
```

Comment:
The new zoom ratio is provided to `onZoom` before the `canvas` is updated and rendered and is not limited by any factors. Sometimes the event's `zoomValue` is out of bounds before the canvas has reached the min/max zoom. 

E.g. The minZoom is 0.5, the canvas is at 0.6, and the zoom increment is -0.2. The desired zoom must be set to the minimum 0.5 rather than remain at 0.6. 

This short circuits the zoom event and triggers a new zoom event with the minimum or maximum zoom if the above scenario occurs.